File: app/src/main/java/com/paycraft/ticketbooker/utils/EditTextDatePicker.java
package com.paycraft.ticketbooker.utils;

import android.app.DatePickerDialog;
import android.content.Context;
import android.view.View;
import android.widget.DatePicker;

import com.google.android.material.textfield.TextInputEditText;

import java.util.Calendar;
import java.util.TimeZone;

public class EditTextDatePicker implements View.OnClickListener, DatePickerDialog.OnDateSetListener {
    private final TextInputEditText _editText;
    private int _day;
    private int _month;
    private int _birthYear;
    private final Context _context;



    public EditTextDatePicker(Context context,TextInputEditText _editText)
    {
        this._editText = _editText;
        this._editText.setOnClickListener(this);
        this._context = context;
    }

    @Override
    public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
        _birthYear = year;
        _month = monthOfYear;
        _day = dayOfMonth;

        updateDisplay();
    }
    @Override
    public void onClick(View v) {
        Calendar calendar = Calendar.getInstance(TimeZone.getDefault());
        calendar.add(Calendar.YEAR, -18);
        DatePickerDialog dialog = new DatePickerDialog(_context, this,
                calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH),
                calendar.get(Calendar.DAY_OF_MONTH));

        dialog.getDatePicker().setMaxDate(calendar.getTimeInMillis());
        calendar.add(Calendar.YEAR, -82);
        dialog.getDatePicker().setMinDate(calendar.getTimeInMillis());
        dialog.show();

    }

    // updates the date in the birth date EditText
    private void updateDisplay() {

        _editText.setText(new StringBuilder()
                // Month is 0 based so add 1
                .append(_day).append("/").append(_month + 1).append("/").append(_birthYear).append(" "));
    }
}

File: app/src/main/java/com/paycraft/ticketbooker/utils/EditTextDatePicker.kt
package com.paycraft.ticketbooker.utils

import android.app.DatePickerDialog
import android.content.Context
import android.view.View
import android.widget.DatePicker

import com.google.android.material.textfield.TextInputEditText

import java.util.Calendar
import java.util.TimeZone

class EditTextDatePicker(private val _context: Context, private val _editText: TextInputEditText) : View.OnClickListener, DatePickerDialog.OnDateSetListener {
    private var _day: Int = 0
    private var _month: Int = 0
    private var _birthYear: Int = 0


    init {
        this._editText.setOnClickListener(this)
    }

    override fun onDateSet(view: DatePicker, year: Int, monthOfYear: Int, dayOfMonth: Int) {
        _birthYear = year
        _month = monthOfYear
        _day = dayOfMonth

        updateDisplay()
    }

    override fun onClick(v: View) {
        val calendar = Calendar.getInstance(TimeZone.getDefault())
        calendar.add(Calendar.YEAR, -18)
        val dialog = DatePickerDialog(_context, this,
                calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH),
                calendar.get(Calendar.DAY_OF_MONTH))

        dialog.datePicker.maxDate = calendar.timeInMillis
        calendar.add(Calendar.YEAR, -82)
        dialog.datePicker.minDate = calendar.timeInMillis
        dialog.show()

    }

    // updates the date in the birth date EditText
    private fun updateDisplay() {

        _editText.setText(StringBuilder()
                // Month is 0 based so add 1
                .append(_day).append("/").append(_month + 1).append("/").append(_birthYear).append(" "))
    }
}